mobile/src/full/java/org/openhab/habdroid/ui/MapViewHelper.java

```diff
@@ -186,17 +187,11 @@ private static void setMarker(GoogleMap map, LatLng position, Item item,
             map.addMarker(marker).setTag(item);
         }
 
-        private static LatLng parseLocation(ParsedState state) {
-            String location = state != null ? state.asString() : null;
-            String[] splitState = location != null ? location.split(",") : null;
-            if (splitState != null && splitState.length == 2) {
-                try {
-                    return new LatLng(Float.valueOf(splitState[0]), Float.valueOf(splitState[1]));
-                } catch (NumberFormatException e) {
-                    // ignored
-                }
-            }
-            return null;
+        private static LatLng toLatLng(ParsedState state) {
+            Location location = state != null ? state.asLocation() : null;
+            return state != null
```

Comment:
Need to check for location == null here.

Reply:
Oops. It's `!=` actually, but thanks, will fix this.